src/main.rs

```diff
@@ -75,11 +95,36 @@ fn ci_env_set() -> bool {
     env::var("CI").is_ok()
 }
 
+fn user_repo_from_url(url: Url) -> Result<(String, String), String> {
+    let path = match url.path() {
+        Some(path) => path,
+        None => return Err("URL should contain user and repository".into()),
+    };
+
+    let user = path[0].clone();
+    let repo = match path[1].rfind(".git") {
+        None => path[1].clone(),
+        Some(suffix_pos) => {
+            let valid_pos = path[1].len() - 4;
+            if valid_pos == suffix_pos {
+                let path = &path[1][0..suffix_pos];
+                path.into()
+            } else {
+                return Err(".git suffix not at the end of URL".into())
```

Comment:
What about `URL does not point to a git repository (Example: http://example.com/foo.git)`?